docs/content/en/docs/documentation/reconciler.md

```diff
@@ -169,3 +169,117 @@ You can specify the name of the finalizer to use for your `Reconciler` using the
+  private final PrimaryResourceCache<StatusPatchPrimaryCacheCustomResource> cache =
+      new PrimaryResourceCache<>(
+          (statusPatchCacheCustomResourcePair, statusPatchCacheCustomResource) ->
+              statusPatchCacheCustomResource.getStatus().getValue()
+                  >= statusPatchCacheCustomResourcePair.afterUpdate().getStatus().getValue());
+
+  @Override
+  public UpdateControl<StatusPatchPrimaryCacheCustomResource> reconcile(
+          StatusPatchPrimaryCacheCustomResource primary,
+          Context<StatusPatchPrimaryCacheCustomResource> context) {
+    
+    // cache will compare the current and the cached resource and return the more recent. (And evict the old)   
+    primary = cache.getFreshResource(primary);
+      
+    // omitted logic  
+      
+    var freshCopy = createFreshCopy(primary);
+    
+    freshCopy.getStatus().setValue(statusWithState());
+
+    var updated =
+            PrimaryUpdateAndCacheUtils.ssaPatchAndCacheStatus(primary, freshCopy, context, cache);
+      
+    return UpdateControl.noUpdate();
+  }
+
+  @Override
+  public DeleteControl cleanup(
+          StatusPatchPrimaryCacheCustomResource resource,
+          Context<StatusPatchPrimaryCacheCustomResource> context)
+          throws Exception {
+    // cleanup the cache on resource deletion  
+    cache.cleanup(resource);
+    return DeleteControl.defaultDelete();
+  }
+  
+```
+
+[`PrimaryResourceCache`](https://github.com/operator-framework/java-operator-sdk/blob/main/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/support/PrimaryResourceCache.java)
+is designed for this purpose. As shown in the example above, it is up to you to provide a predicate to determine if the
+resource is more recent than the one available. In other words, when to evict the resource from the cache. Typically, as
+shown in
+the [integration test](https://github.com/operator-framework/java-operator-sdk/blob/main/operator-framework/src/test/java/io/javaoperatorsdk/operator/baseapi/statuscache/primarycache)
+you can have a counter in status to check on that.
+
+Since all of this happens explicitly, you cannot use this approach for managed dependent resources and workflows and
+will need to use the unmanaged approach instead. This is due to the fact that managed dependent resources always get
+their associated primary resource from the underlying informer event source cache.
+
+#### Additional remarks
+
+As shown in the integration tests, there is no optimistic locking used when updating the
+[resource](https://github.com/operator-framework/java-operator-sdk/blob/main/operator-framework/src/test/java/io/javaoperatorsdk/operator/baseapi/statuscache/internal/StatusPatchCacheReconciler.java#L41)
+(in other words `metadata.resourceVersion` is set to `null`). This is desired since you don't want the patch to fail on
+update.
+
+In addition, you can configure the Fabric8 client retry. 
```

Comment:
This probably needs to be explained more and, possibly, point to the appropriate documentation.

Reply:
added link